src/odemis/gui/cont/tabs.py

```diff
@@ -4391,7 +4395,475 @@ def _on_align_move_done(self, f=None):
+                                         panel.beam_shift_btn_p_aligner_x:
+                                             lambda evt: self._sem_alignment_controls("x", 1, evt),
+                                         panel.beam_shift_btn_m_aligner_y:
+                                             lambda evt: self._sem_alignment_controls("y", -1, evt),
+                                         panel.beam_shift_btn_p_aligner_y:
+                                             lambda evt: self._sem_alignment_controls("y", 1, evt),
+                                         }
+        flm_aligner_control_functions = {panel.flm_align_btn_m_aligner_x:
+                                             lambda evt: self._flm_alignment_controls("x", -1, evt),
+                                         panel.flm_align_btn_p_aligner_x:
+                                             lambda evt: self._flm_alignment_controls("x", 1, evt),
+                                         panel.flm_align_btn_m_aligner_y:
+                                             lambda evt: self._flm_alignment_controls("y", -1, evt),
+                                         panel.flm_align_btn_p_aligner_y:
+                                             lambda evt: self._flm_alignment_controls("y", 1, evt),
+                                         panel.flm_align_btn_m_aligner_z:
+                                             lambda evt: self._flm_alignment_controls("z", -1, evt),
+                                         panel.flm_align_btn_p_aligner_z:
+                                             lambda evt: self._flm_alignment_controls("z", 1, evt),
+                                         }
+
+        self._combined_aligner_control_functions = {**z_aligner_control_functions,
+                                                    **sem_aligner_control_functions,
+                                                    **flm_aligner_control_functions}
+
+        # Bind alignment control buttons to defined control functions
+        for btn, function in self._combined_aligner_control_functions.items():
+            btn.Bind(wx.EVT_BUTTON, function)
+
+        # Vigilant attribute connectors for the slider
+        self._step_size_controls_va_connector = VigilantAttributeConnector(tab_data.step_size,
+                                                                           self.panel.controls_step_size_slider,
+                                                                           events=wx.EVT_SCROLL_CHANGED)
+
+        # Bind the buttons to the button move to custom alignment
+        def __on_click_custom_alignment(evt):
+            # First move the stage then the objective
+            future_stage_move = self._stage.moveAbs(self._stage.getMetadata()[model.MD_FAV_POS_ACTIVE])
+            move_alginer = lambda future: self._aligner.moveAbs(self._aligner.getMetadata()[model.MD_FAV_POS_ACTIVE])
```

Comment:
Since these movements where considered to be always small (because we are already at least in the factory alignment). We (Caspar, Daan and I) decided that we could do the moves in all direction all at once. Of course if something fails the expection should still be logged. can I just add a try - except statement for that or isn't that enough?

Reply:
try/except work... if you put them around the `future.result()`. You can also use future.exception() to directly check, which might be easier in a callback function.
For an example, you can look at gui.model.ActuatorGUIData._on_axis_move_done()

Reply:
I'm a bit confused on how to implement this without having the entire GUI geing stuck on waiting while the move is finished. Can we look at this together?